src/backend/executor/execMain.c

```diff
@@ -648,6 +650,18 @@ ExecCheckOneRelPerms(RTEPermissionInfo *perminfo)
 	requiredPerms = perminfo->requiredPerms;
 	Assert(requiredPerms != 0);
 
+	/* 
+	 * Some extension may want to avoid permission check for some special
+	 * cases. For example, Babelfish temp table is implemented using ENR
+	 * which is not shared with parallel worker and parallel operations
+	 * are not allowed for temp table in Postgres. So give extension a
+	 * chance to skip permission check for such use cases.
+	 */
+	if (IsBabelfishParallelWorker() &&
+		ExecCheckOneRelPerms_hook &&
```

Comment:
can we put extension hook check first to be more safe

Reply:
what difference will it make?

Reply:
Shouldn't the parallel worker check be in the extension?